bokehjs/src/coffee/core/has_props.coffee

```diff
@@ -386,9 +387,11 @@ export class HasProps extends Backbone.Model
       if need_invalidate
         @document._invalidate_all_models()
 
-      @document._notify_change(@, attr, old, new_)
+      if not options?.notify? or options.notify == true
+        @document._notify_change(@, attr, old, new_)
```

Comment:
@havocp I think this is the main thing I need your input on. When a ``ModelChangeEvent`` is received by the frontend it now sets a new ``notify=false`` option which gets passed through to here. This has the effect of preventing all callbacks on the Document from being notified about the change. This results in a fairly large speedup because it doesn't have to compare all the events that are received only to then suppress them anyway.

My question is, is this safe? What other types of callbacks apart from a ClientSession might be attached to a document? Is there a case where you do actually want a received ModelChangeEvent to trigger these callbacks?